src/contentmanager.cpp

```diff
@@ -232,36 +234,24 @@ void ContentManager::setCurrentCategoryFilter(QString category)
     emit(filterParamsChanged());
 }
 
-#define CATALOG_HOST "library.kiwix.org"
-#define CATALOG_PORT 80
-#define CATALOG_URL "library.kiwix.org"
 void ContentManager::updateLibrary() {
     if (m_local) {
+        emit(pendingRequest(false));
         emit(booksChanged());
         return;
     }
-    QUrlQuery query;
-    if (m_currentLanguage != "*") {
-        query.addQueryItem("lang", m_currentLanguage);
-    }
-    query.addQueryItem("count", QString::number(0));
-    if (m_categoryFilter != "all") {
-        query.addQueryItem("tag", m_categoryFilter);
-    }
-    QUrl url;
-    url.setScheme("http");
-    url.setHost(CATALOG_HOST);
-    url.setPort(CATALOG_PORT);
-    url.setPath("/catalog/search");
-    url.setQuery(query);
-    qInfo() << "Downloading" << url.toString(QUrl::FullyEncoded);
-    m_remoteLibrary = kiwix::Library();
-    kiwix::Manager manager(&m_remoteLibrary);
     try {
-        auto allContent = kiwix::download(url.toString(QUrl::FullyEncoded).toStdString());
-        manager.readOpds(allContent, CATALOG_URL);
+        emit(pendingRequest(true));
+        m_remoteLibrary = kiwix::Library();
+        m_remoteLibraryManager.doUpdate(m_currentLanguage, m_categoryFilter);
     } catch (runtime_error&) {}
-    emit(booksChanged());
+}
+
+void ContentManager::updateRemoteLibrary(const QString& content) {
+    kiwix::Manager manager(&m_remoteLibrary);        
+    manager.readOpds(content.toStdString(), "library.kiwix.org");
```

Comment:
We should keep CATALOG_URL here. (And keep its definition in `src/contentmanagaer.cpp` instead of moving it in `src/opdsrequestmanager.cpp` where it is never used.